modules/core/src/lib/deck-renderer.js

```diff
@@ -88,15 +101,56 @@ export default class DeckRenderer {
     return redraw;
   }
 
+  finalize() {
+    if (this.screenBuffer) {
+      this.screenBuffer.delete();
+      this.screenBuffer = null;
+    }
+    if (this.offscreenBuffer) {
+      this.offscreenBuffer.delete();
+      this.offscreenBuffer = null;
+    }
+  }
+
   // Private
   prepareEffects(effects) {
     const effectProps = {};
+    const postProcessEffects = [];
```

Comment:
Perf: instead of building a new array on each render, you can just use a boolean flag here:

```js
let hasPostProcessEffects = false;
...
if (hasPostProcessEffects) {
  this.prepareRenderBuffers();
}
```

And `renderPostProcessEffects(postProcessEffects)` becomes `postRender(effects)`. This has the following advantages:

- Performing `instanceof` check again in a for loop is very cheap, while garbage collection has a cost.
- Introducing a generic lifecycle method, instead of something specific for `PostProcessEffect`, allows this class to accommodate future effect types without constantly adding new APIs.